packages/monitor-v2/src/monitor-og/oSnapAutomation.ts

```diff
@@ -242,6 +242,29 @@ const filterPotentialProposals = (
   });
 };
 
+// Filters out all Snapshot proposals that cannot be proposed due to blocking on-chain proposals. This is done by
+// matching safe and proposed transactions.
+const filterUnblockedProposals = async (
+  potentialProposals: SnapshotProposalExpanded[],
+  onChainProposals: TransactionsProposedEvent[],
+  params: MonitoringParams
+): Promise<SnapshotProposalExpanded[]> => {
+  // Filter out all on-chain proposals that have been executed since they cannot block new proposals.
+  const unexecutedProposals = await filterUnexecutedProposals(onChainProposals, params);
+
+  return potentialProposals.filter((potentialProposal) => {
+    // Unexecuted proposals with the same safe and matching transactions would block the new proposal.
+    const blockingOnChainProposals = unexecutedProposals.filter((unexecutedProposal) => {
+      return (
+        isMatchingSafe(potentialProposal.safe, params.chainId, unexecutedProposal.address) &&
+        onChainTxsMatchSnapshot(unexecutedProposal, potentialProposal.safe)
+      );
```

Comment:
Maybe I'm missing something, but where are we checking that the txns from the potential proposal matches the unexecuted proposal?

Reply:
[onChainTxsMatchSnapshot](https://github.com/UMAprotocol/protocol/blob/6aaf13b75396d94288a2856502903a33900f2922/packages/monitor-v2/src/monitor-og/SnapshotVerification.ts#L372) should check the number of tx matches and that to/data/value/operation properties match on each tx